Remove deprecated functions from importers

dsenalik opened the pull request:
# Tripal 4 Core Dev Task

### Closes #2295 

### ~~Depends on #2221~~ (Merged)

### ~~Depends on #2294~~ (Merged)

## Description

PR #2294 will deprecate a number of `chado_` api functions inherited from Tripal 3.

This PR replaces them in various importers.

## Testing?

~~Before testing `git merge origin/tv4g2-1341-deprecateChadoQueryAPI` so that deprecated functions show up in testing.~~
Automated tests should now catch any missed deprecations

Run each of the importers to verify that I have not broken anything.
To test obo importer on a local file you can use `modules/contrib/tripal/tripal_chado/tests/fixtures/obo_loader/test_EDAM.obo`

dsenalik committed:
remove deprecated functions: newick importer

dsenalik committed:
remove deprecated chado_get_cvterm() from chado_get_organism_scientific_name()

dsenalik committed:
remove deprecated functions: tree generator

dsenalik committed:
remove deprecated functions: taxonomy importer

dsenalik committed:
remove deprecated functions not covered by tests

dsenalik committed:
remove deprecated functions: ChadoPhylogenyAPITest

dsenalik committed:
Merge remote-tracking branch 'origin/tv4g2-1341-deprecateChadoQueryAPI' into tv4g0-issue2295-remove-deprecated-functions-from-importers

dsenalik committed:
Merge remote-tracking branch 'origin/4.x' into tv4g0-issue2295-remove-deprecated-functions-from-importers

laceysanderson committed:
Merge branch '4.x' into tv4g0-issue2295-remove-deprecated-functions-from-importers

laceysanderson committed:
Merge remote-tracking branch 'origin/tv4g2-1341-deprecateChadoQueryAPI' into tv4g0-issue2295-remove-deprecated-functions-from-importers

laceysanderson committed:
Ensures phpunit will fail when deprecations are encountered on D11 (PHPUnit 11.5). This should not fail on PHPUnit deprecations such as the getMockForAbstractClass().

laceysanderson commented:
I made this PR dependant on #2294 so that we can check the deprecations right now. I also added a setting to PHPUnit 11.5 config to ensure that our tests will fail when deprecations are encountered outside of tests with `#[IgnoreDeprecations]`. This settings should not fail when the deprecations are caused by third-party code such as PHPunit so the long standing mock deprecations (See Issue #2249) can remain unaddressed for now 😅

dsenalik committed:
Merge branch 'tv4g0-issue2295-remove-deprecated-functions-from-importers' of github.com:/tripal/tripal into tv4g0-issue2295-remove-deprecated-functions-from-importers

dsenalik committed:
Merge remote-tracking branch 'origin/4.x' into tv4g0-issue2295-remove-deprecated-functions-from-importers

dsenalik committed:
ignore deprecations organism API

dsenalik committed:
you need a use statement too 🙃

laceysanderson commented on tripal_chado/tests/src/Kernel/api/ChadoPhylogenyAPITest.php:
```diff
@@ -83,15 +85,19 @@ public function testChadoPhylotreeAPIFunctions() {
-    $dbq = chado_insert_record('organism', $org, [], $this->schemaName);
-    $this->assertNotNull($dbq, 'Unable to insert test organism 1.');
-    $organism_ids[0] = $dbq['organism_id'];
+    $query = $this->chado_connection->insert('1:organism');
+    $organism_id = $query->fields($org);
```
```suggestion
    $query = $this->chado_connection->insert('1:organism')
      ->fields($org);
```

laceysanderson commented on tripal_chado/tests/src/Kernel/api/ChadoPhylogenyAPITest.php:
```diff
@@ -83,15 +85,19 @@ public function testChadoPhylotreeAPIFunctions() {
-    $dbq = chado_insert_record('organism', $org, [], $this->schemaName);
-    $this->assertNotNull($dbq, 'Unable to insert test organism 2.');
-    $organism_ids[1] = $dbq['organism_id'];
+    $query = $this->chado_connection->insert('1:organism');
+    $organism_id = $query->fields($org);
```
```suggestion
    $query = $this->chado_connection->insert('1:organism')
      ->fields($org);
```

laceysanderson commented on tripal_chado/src/api/tripal_chado.phylotree.api.php:
```diff
@@ -998,26 +997,25 @@ function chado_phylogeny_lookup_organism_by_name($name, $schema_name = 'chado')
  * @ingroup tripal_phylotree_api
  */
-function chado_phylogeny_get_node_types_vocab($options, $schema_name = 'chado') {
+function chado_phylogeny_get_node_types_vocab($chado, $options, $schema_name = 'chado') {
```
This is not backwards compatible and we have not added deprecation messages indicating the change :see_no_evil:

If this change was made because of the default of schema name for chado then we can either update the importer to pass in the schema name or we can add a section here that if the schema name is not passed in then do not set it or grab the set default schema name.

```suggestion
function chado_phylogeny_get_node_types_vocab($options, $schema_name = 'chado') {
  $chado = \Drupal::service('tripal_chado.database');
  $chado->setSchemaName($schema_name);
```

laceysanderson reviewed (changes requested):
Code review is looking mostly good :-) Just one backwards incompatible change that we should resolve.

dsenalik committed:
Apply suggestion from @laceysanderson

I added the parameter to avoid loading chado again, and it is only called by one other function. However, you are right about compatibility breaking, even if unlikely, so your proposed change is the best solution. Someday this API should be redone anyway.

Co-authored-by: Lacey-Anne Sanderson <lacey.sanderson@usask.ca>

dsenalik committed:
Apply suggestion from @laceysanderson

Co-authored-by: Lacey-Anne Sanderson <lacey.sanderson@usask.ca>

dsenalik committed:
Apply suggestion from @laceysanderson

Co-authored-by: Lacey-Anne Sanderson <lacey.sanderson@usask.ca>

dsenalik committed:
remove compatibility-breaking parameter

dsenalik committed:
revert chado_phylogeny_get_node_types_vocab call in test

laceysanderson committed:
Check for sequence ontology earlier in the form generation.

laceysanderson reviewed (commented):
## Manual Testing

### GFF3 importer.

The change was in insertFeatureSeqs() so I tested 
- using this test fixture `tripal_chado/tests/fixtures/gff3_loader/small_gene_with_fasta.gff`
- imported it with a default srcfeature type of scaffold
- created a scaffold content type and published it
- ✅ The residues and associated metadata showed up as expected
<img width="1065" height="963" alt="Screenshot 2025-09-29 at 12 04 46 PM" src="https://github.com/user-attachments/assets/f1dff3b0-be66-419b-9a05-2c42f7c250f1" />

### Newick Tree Loader

The change here was in the form when the sequence ontology is not present so I tested
- by updating the name of the sequence ontology to `Fred` which did trigger the code... However, apparently the original code was checking too late and other errors appeared. 
- As such I make the code change in commit f73a8e61b5edded2e6653b7037d36a5ad6306fc7 with @dsenalik blessing
- ✅ Now loading the form shows the warning as expected.
<img width="1775" height="1059" alt="Screenshot 2025-09-29 at 12 21 00 PM" src="https://github.com/user-attachments/assets/2f2657f8-e607-43e9-975f-b074bfdfd4f3" />

laceysanderson committed:
Indicate cvterm_id to be selected field.

laceysanderson reviewed (changes requested):
## Manual Testing

### OBO Importer

The change here was in the stanza caching. This should be impacted on most obo loads so I tested by 
- I used the local EDAM subset suggested by Doug
- On the obo imporer I selected "Add a New Ontology", included the path provided by Doug and set the same to "Local EDAM"
- ✅ I then ran the job and confirmed there were no errors as expected. If the changes had not worked, there would have been PDO errors.
   <img width="1394" height="741" alt="Screenshot 2025-09-29 at 1 07 10 PM" src="https://github.com/user-attachments/assets/6e9f8288-80de-4fef-88b3-faeeec127e47" />

### Taxonomy Importer
Changes here were made in the following methods: findOrganism(), addOrganism(), and importRecord(). These will be hit on all jobs for this importer that has at least one organism. As such I tested by
- Created the organism `Daucus muricatus` which matches one of the ids in the next step through the Add Organism form (to test find when one exists).
- Imported the following ids `48026 48106 79170 79196 79200 180796` using the "NCBI Taxonomy Importer" form and selected "Import details for existing species"
- When I ran the job it failed with the following error showing the updated query in addOrganism() did not work.
   <img width="1414" height="476" alt="Screenshot 2025-09-29 at 1 22 06 PM" src="https://github.com/user-attachments/assets/d86c5727-cc07-40ba-b644-1bd37694c0ba" />
- ❌ I fixed this by adding the field to be selected to one of the queries (see commit 37bbd1795181f3ca4cbb04c9d771f49be013451f) but it still failed as follows. Note: While I did add the one organism ahead of time and it does have an organism_id of 5, it did not have a dbxref_id assigned via organism_dbxref before the import.
   ```sql
   sitedb=# select * from teapot.organism;
    organism_id | abbreviation |  genus   |  species   | common_name | infraspecific_name | type_id | comment 
   -------------+--------------+----------+------------+-------------+--------------------+---------+---------
              1 |              | Tripalus | databasica |             |                    |         | 
              5 |              | Daucus   | muricatus  |             |                    |         | 
   (2 rows)

   sitedb=# select * from teapot.organism_dbxref;
    organism_dbxref_id | organism_id | dbxref_id 
   --------------------+-------------+-----------
   (0 rows)
   ```

   <img width="1411" height="328" alt="Screenshot 2025-09-29 at 1 39 41 PM" src="https://github.com/user-attachments/assets/16c8fc6b-ca16-46d8-8565-286d3272b123" />

dsenalik committed:
check before associating dbxref

dsenalik committed:
Need fetchField()

laceysanderson reviewed (approved):
## Manual Testing

✅ GFF3 Importer
✅ Newick Tree Loader
✅ OBO Importer
✅ Taxonomy importer
✅ Tree Generator
✅ FASTA Importer

### Taxonomy Importer
Changes here were made in the following methods: findOrganism(), addOrganism(), and importRecord(). These will be hit on all jobs for this importer that has at least one organism. As such I tested by
- Created the organism `Daucus muricatus` which matches one of the ids in the next step through the Add Organism form (to test find when one exists).
- Imported the following ids `48026 48106 79170 79196 79200 180796` using the "NCBI Taxonomy Importer" form and selected "Import details for existing species"
- When I ran the job, it completed without error as expected :-)
- ✅ After finding the new fields and re-publishing, I was able to confirm all the details are present both on the pre-existing `Daucus muricatus` and one other organism.
   <img width="1157" height="1257" alt="image" src="https://github.com/user-attachments/assets/ce711eaf-64a3-492c-a1bc-c5423d2ca1b1" />
   <img width="1157" height="1258" alt="image" src="https://github.com/user-attachments/assets/85530ce1-2afd-4f9a-b085-116896f01bed" />

### Taxonomy Tree Generator

Changes were made in the run(), retrieveLineage(), removeTree(), and queryPhylonode(). I believe these changes will be best fit by regenerating an existing tree and trying to generate a tree that doesn't have an appropriate root organism.
- Tried to generate tree where the root taxon was "fabaceae" despite there being no organisms in the site for this family.
- ✅ Confirmed that the job ran with the expected error and that the initial record added by the importer was removed leaving an empty `phylotree` table.
   <img width="1280" height="313" alt="Screenshot 2025-09-29 at 3 44 03 PM" src="https://github.com/user-attachments/assets/31ec5dcc-8458-4a38-87c2-ac6b2e0aec2c" />
- Used the importer to generate a tree for `Daucus` genus which there are 4 organism in the site.
- ✅ the job ran without error, confirmed there were 4 organism which should be included and 2 which should not. Selecting from the database shows there are the expected phylonode records 🎉 
   <img width="1330" height="585" alt="Screenshot 2025-09-29 at 3 54 01 PM" src="https://github.com/user-attachments/assets/a96f9941-b11a-4f9e-8344-98be48ff518a" />

### FASTA Importer

Changes are made to loadFastaFeature(), specifically with reference to updating an existing feature. As such we are testing by
- Import features using the GFF3 importer and `/var/www/drupal/web/modules/contrib/tripal/tripal_chado/tests/fixtures/gff3_loader/gff_Citrus_sinensis-orange1.1g015632m.g.gff3`
- Import the sequence for orange1.1g015632m using the FASTA importing using the `/var/www/drupal/web/modules/contrib/tripal/tripal_chado/tests/fixtures/fasta_loader/Citrus_sinensis-orange1.1g015632m.g.fasta`
- ✅ I confirmed that the sequence was imported and assigned to the residues field of the existing gene.
   <img width="1157" height="1216" alt="image" src="https://github.com/user-attachments/assets/b670418d-803a-468a-8e8b-16ee149e9437" />
